client/state/notices/reducer.js

```diff
@@ -31,6 +32,16 @@ export const items = createReducer( {}, {
 
 		return omit( state, noticeId );
 	},
+	[ NOTICE_CLEAR ]: ( state ) => {
+		return reduce( state, ( memo, notice, noticeId ) => {
```

Comment:
Good question! I kept it like this to be "compatible" with the `ROUTE_SET` action. And that made sense for my use case. I'm not sure if clearing all the notices including persistent ones could be useful. I'll add a comment for now to the action creator. I think a parameter could easily be added later if needed.